### PR TITLE
Simplified logging middleware; Fields are now "any" type; Moved logging providers to examples only.

### PR DESCRIPTION
**Pull request by bwplotka**
Fixes https://github.com/grpc-ecosystem/go-grpc-middleware/issues/517

* Simpler interface, only one method. This allows adapter to be ultra simple, so moved them to examples only (!).
* Fields are string, values are any.
* "More" slog compatibility.

I did not go for using slog natively as it's still experimental (see bigger discussion: https://gophers.slack.com/archives/C029RQSEE/p1679860885943619)

### CODE REVIEW

File: interceptors/logging/slog.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// Any system can map them to another numbering scheme if it wishes.
+//
+// First, we wanted the default level to be Info, Since Levels are ints, Info is
+// the default value for int, zero.
+//
+
+// Second, we wanted to make it easy to use levels to specify logger verbosity.
+// Since a larger level means a more severe event, a logger that accepts events
+// with smaller (or more negative) level means a more verbose logger. Logger
+// verbosity is thus the negation of event severity, and the default verbosity
```
Comment (yashrsharma44) [inline interceptors/logging/slog.go:27]: Should using "inverse" would be a better option? Negation implies that our definition is tied up in the implementation(we are negating in this case), however using inverse means that lower the log level(enum) included, more verbose our logging is, which is simple to understand.

File: interceptors/logging/logging.go [hunk 2]
```diff
@@ -119,70 +106,69 @@ func (f Fields) AppendUnique(add Fields) Fields {
...
+func InjectLogField(ctx context.Context, key string, val any) context.Context {
+	return InjectFields(ctx, Fields{key, val})
 }
 
-// Logger is unified interface that we used for all our interceptors. Official implementations are available under
-// provider/ directory as separate modules.
+// Logger requires Log method, similar to experimental slog, allowing logging interceptor to be interoperable. Official
+// adapters for popular loggers are in `provider/` directory (separate modules). It's totally ok to copy simple function
+// implementation over.
+// TODO(bwplotka): Once slog is official, we could use slog method directly. Currently level is copied over, so we don't
```
Comment (yashrsharma44) [inline interceptors/logging/logging.go:162]: 👍

File: interceptors/logging/examples/kit/example_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	"google.golang.org/grpc"
+)
+
+// InterceptorLogger adapts go-kit logger to interceptor logger.
+// This code is simple enough to be copied and not imported.
+func InterceptorLogger(l log.Logger) logging.Logger {
+	return logging.LoggerFunc(func(_ context.Context, lvl logging.Level, msg string, fields ...any) {
+		largs := append([]any{"msg", msg}, fields...)
+		switch lvl {
+		case logging.LevelDebug:
```
[the remark this answers is not in the document]
Comment (bwplotka) [reply interceptors/logging/examples/kit/example_test.go:22]: What is ERROR? ERROR was replaced by LevelError and comon levels from slog there.

File: interceptors/logging/slog.go (hunk 1 above)
Comment (bwplotka) [reply interceptors/logging/slog.go:27]: I just copied all of this from slog, so there is less maintenance on our side (:  So if you don't like this you can talk to slog authors 🙈  (https://pkg.go.dev/golang.org/x/exp/slog#pkg-overview)... I feel we can use whatever, so I thought we could use what others most likely start doing more.

File: interceptors/logging/examples/kit/example_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (yashrsharma44) [reply interceptors/logging/examples/kit/example_test.go:22]: Okay, I meant we were using `logging.ERROR` before, but if slog uses that - [link](https://github.com/grpc-ecosystem/go-grpc-middleware/blob/4402c19b3d262c7c77c3aa771ec3d322247b68f6/interceptors/logging/slog.go#L42), then let's use that convention.

**Review by yashrsharma44 — APPROVED** [review]

**Pull request merged**